l10n_it_partially_deductible_vat/models/account.py

```diff
@@ -0,0 +1,344 @@
+        invoice = recs.env['account.invoice'].browse(invoice_id)
+        return recs.compute(invoice)
+
+
+class AccountTaxCode(models.Model):
+
+    _inherit = 'account.tax.code'
+
+    base_tax_ids = fields.One2many('account.tax', 'base_code_id', 'Base Taxes')
+    tax_ids = fields.One2many('account.tax', 'tax_code_id', 'Taxes')
+    ref_base_tax_ids = fields.One2many(
+        'account.tax', 'ref_base_code_id', 'Ref Base Taxes')
+    ref_tax_ids = fields.One2many(
+        'account.tax', 'ref_tax_code_id', 'Ref Taxes')
+
+
+class AccountInvoiceLine(models.Model):
+
+    _inherit = 'account.invoice.line'
+
+    @api.model
+    def move_line_get(self, invoice_id):
+        inv = self.env['account.invoice'].browse(invoice_id)
+        currency = inv.currency_id.with_context(date=inv.date_invoice)
+        company_currency = inv.company_id.currency_id
+        res = []
+        for line in inv.invoice_line:
+            mres = self.move_line_get_item(line)
+            mres['invl_id'] = line.id
+            res.append(mres)
+            tax_code_found = False
+            taxes = line.invoice_line_tax_id.compute_all(
+                (line.price_unit * (1.0 - (line.discount or 0.0) / 100.0)),
+                line.quantity, line.product_id, inv.partner_id)['taxes']
+            for tax in taxes:
+                if inv.type in ('out_invoice', 'in_invoice'):
+                    tax_code_id = tax['base_code_id']
+                    # Changed for https://github.com/OCA/OCB/commit/
+                    # d83befdb0f4cdc34f1d2aad7f91f07efe7d3bccf
+                    # Else it would put a wrong amount in base tax
+                    if self.env['account.tax'].browse(
+                            tax['id']).include_base_amount:
+                        tax_amount = tax['price_unit'] * line.quantity * \
+                            tax['base_sign']
+                    else:
+                        tax_amount = line.price_subtotal * tax['base_sign']
+                else:
+                    tax_code_id = tax['ref_base_code_id']
+                    if self.env['account.tax'].browse(
+                            tax['id']).include_base_amount:
+                        tax_amount = tax['price_unit'] * line.quantity * \
+                            tax['ref_base_sign']
+                    else:
+                        tax_amount = line.price_subtotal * tax['base_sign']
+                    # End change
+
+                if tax_code_found:
+                    if not tax_code_id:
+                        continue
+                    res.append(dict(mres))
+                    res[-1]['price'] = 0.0
+                    res[-1]['account_analytic_id'] = False
+                elif not tax_code_id:
+                    continue
+                tax_code_found = True
+
+                res[-1]['tax_code_id'] = tax_code_id
+                res[-1]['tax_amount'] = currency.compute(
+                    tax_amount, company_currency)
+
+        return res
```

Comment:
Conviene fare override chiamando ```super```, altrimenti la catena delle dipendenze verrebbe rotta.
In questo caso ```res``` contiene i valori che si vogliono modificare, quindi sarebbe sufficiente, dopo aver chiamato ```super```, andare a modificare direttamente i valori di ```res```